src/include/abti_mutex.h

```diff
@@ -8,42 +8,42 @@
 
 /* Inlined functions for Mutex */
 
-#define ABTI_PTR_SPINLOCK(ptr)                          \
-    while (ABTD_atomic_cas_uint32(ptr, 0, 1) != 0) {    \
-        while (*(volatile uint32_t *)(ptr) != 0) {      \
-        }                                               \
+#define ABTI_PTR_SPINLOCK(ptr)                             \
+    while (!ABTD_atomic_bool_cas_weak_uint32(ptr, 0, 1)) { \
+        while (*(volatile uint32_t *)(ptr) != 0) {         \
```

Comment:
ambiguous non-atomic access of an atomic variable

Reply:
This commit just replaces all atomic operations with new ones.
These macros are removed in the next commit.

Reply:
whitespace cleanup